index.html

```diff
@@ -90,54 +90,178 @@
 
 <div class=issue data-number=8></div>
 
+### The <dfn>Sensors</dfn> Interface
 
+The <a>Sensors</a> interface represents a container for a list of <a>Sensor</a> objects. It is exposed on Window and Workers as the `Window.sensors` and `WorkerGlobalScope.sensors` attribute respectively.
 
+<pre class="idl">
+[Constructor, Exposed=(Window,Worker)]
+interface Sensors {
+  Promise&lt;sequence&lt;Sensor&gt;&gt; matchAll(optional MatchAllOptions options);
+};
+
+partial interface Window {
+  [SameObject] readonly attribute Sensors sensors;
+};
+
+partial interface WorkerGlobalScope {
+  [SameObject] readonly attribute Sensors sensors;
+};
+
+dictionary MatchAllOptions {
+  DOMString type;
+};
+</pre>
+
+#### <dfn>Sensors.matchAll</dfn>
+
+<div class=issue data-number=26></div>
+
+Returns a promise which resolves to an array of <a>Sensor</a> objects representing all available local(?) sensors.
+
+<div class=issue data-number=28></div>
+<div class=issue data-number=7></div>
+
+<pre class="example highlight">
+sensors.matchAll({ type: "proximity", position: "rear" }).then(function(sensors) {
+    var sensor = sensors[0];
+    if (!sensor) return;
+    // sensor.consructor === ProximitySensor;
+    var obs = new SensorObserver(sensor, { /* ... */ });
+    obs.onchange = dostuff;
+});
+</pre>
 
 ### The <dfn>Sensor</dfn> Interface
 
+The `Sensor` interface is a lightweight object that represents an actual physical sensor. Concrete sensor implementation will need to subclass this.
+
+If no `identifier` is given, the Constructor uses the identifier of the default sensor of that type. This is to the catter for the common case of having a single sensor of a given type on a device.
+
+<pre class="example highlight javascript">
+
+let sensor = new GeolocationSensor();
+sensor.isDefault; // returns true;
+
+</pre>
+
 <pre class="idl">
-[Constructor(SensorInit sensorInitDict), Exposed=(Window,Worker)]
-interface Sensor : EventTarget {
-  static Promise&lt;sequence&lt;DOMString&gt;&gt; getAll();
+[Constructor(optional DOMString identifier, optional SensorInit sensorInitDic), Exposed=(Window,Worker)]
+interface Sensor {
+    readonly attribute DOMString identifier;
+    readonly attribute boolean isDefault;
+    SensorObserver getObserver(optional SensorObserverOptions sensorObserverOptions);
+};
+
+dictionary SensorInit {
+  boolean isDefault;
+};
+</pre>
+
+#### <dfn>Sensor.identifier</dnf>
+
+Returns the identifier of the sensor. This is an opaque DOMString.
+
+#### <dfn>Sensor.isDefault</dnf>
+
+Returns true if the sensor is the default sensor of that type on the device, false otherwise.
+
+#### <dfn>Sensor.getObserver</dnf>
+
+Returns a new <a>SensorObserver</a> object attached to this particular sensor.
```

Comment:
Why do we need a factory? This might confuse people who expect a singleton. 
```
x.getObserver() === x.getObserver()
```

Why not just have SensorObserver be constructable and follow the model used by `MutationObserver`?